shipyard_blueprints/portable/shipyard_portable/portable.py

```diff
@@ -0,0 +1,152 @@
+        self.headers = {"authorization": f"Bearer {self.access_token}"}
+
+        super().__init__(access_token)
+
+    def connect(self):
+        """
+        Connects to the Portable API.
+
+        Returns:
+            int: 0 if the connection is successful, 1 otherwise.
+        """
+        response = requests.get(self.flow_url, headers=self.headers)
+        if response.status_code == 200:
+            logger.info("Successfully connected to Portable API")
+            return 0
+        else:
+            logger.info(f"Status code {response.status_code}")
+            logger.info(f"Response text {response.text}")
+
+            logger.error(
+                "Failed to connect to Portable API. Ensure that the API key provided is correct"
+            )
+            return 1
+
+    def trigger_sync(self, flow_id: str) -> Dict[str, str]:
+        """
+        Triggers a sync for a specific flow.
+
+        Parameters:
+        - flow_id (str): The ID of the flow to trigger sync for.
+
+        Returns:
+        - Dict[str, str]: A dictionary containing the response from the API.
+
+        Raises:
+        - UnauthorizedError: If the API returns a 401 status code.
+        - BadRequestError: If the API returns a 400 status code.
+        - ExitCodeException: If the API returns any other status code.
+
+        """
+        try:
+            flow_url = f"{self.flow_url}/{flow_id}/run"
+            response = requests.post(flow_url, headers=self.headers)
+            response.raise_for_status()
+
+            if response.status_code == 201:
+                logger.info(f"Successfully triggered sync for flow {flow_id}")
+
+        except Exception as e:
+            if response.status_code == 401:
+                raise UnauthorizedError(response.text)
+
+            elif response.status_code == 400:
+                raise BadRequestError(response.text)
+            else:
+                logger.debug(f"Status code {response.status_code}")
+                raise ExitCodeException(
+                    f"Failed to trigger sync for flow {flow_id} due to an expected error. Message from the API: {response.text}",
+                    Etl.EXIT_CODE_UNKNOWN_ERROR,
+                )
+        else:
+            return response.json()
+
+    def get_sync_status(self, flow_id: str) -> Dict[str, str]:
+        """
+        Retrieves the synchronization status for a given flow.
+
+        Args:
+            flow_id (str): The ID of the flow.
+
+        Returns:
+            Dict[str, str]: A dictionary containing the synchronization status.
+
+        Raises:
+            UnauthorizedError: If the request is unauthorized (status code 401).
+            BadRequestError: If the request is invalid (status code 400).
+            ExitCodeException: If an unexpected error occurs while fetching the status.
+        """
+        try:
+            flow_url = f"{self.flow_url}/{flow_id}/status"
+            response = requests.get(flow_url, headers=self.headers)
```

Comment:
suggestions:
flow_url is not reused so I would've
`response = requests.get(f"{self.flow_url}/{flow_id}/status", headers=self.headers)`

also would've made a `_handle_error` function since 
```
  except Exception as e:
            if response.status_code == 401:
                raise UnauthorizedError(response.text)
            elif response.status_code == 400:
                raise BadRequestError(response.text)
            else:
```
seems to be reused. It is a small code so not needed at this point but still worth considering